packages/kbn-expandable-flyout/src/components/settings_menu.tsx

```diff
@@ -59,48 +61,62 @@ const FLYOUT_TYPE_PUSH_TOOLTIP = i18n.translate('expandableFlyout.settingsMenu.p
   defaultMessage: 'Displays the flyout next to the page',
 });
 
-interface SettingsMenuProps {
+export interface FlyoutCustomProps {
   /**
-   * Current flyout type
+   * Hide the gear icon and settings menu if true
    */
-  flyoutTypeProps: {
-    /**
-     * 'push' or 'overlay'
-     */
-    type: EuiFlyoutProps['type'];
-    /**
-     * Callback to change the flyout type
-     */
-    onChange: (type: EuiFlyoutProps['type']) => void;
+  hideSettings?: boolean;
+  /**
+   * Control if the option to render in overlay or push mode is enabled or not
+   */
+  pushVsOverlay?: {
     /**
-     * Disables the button group for flyout where the option shouldn't be available
+     * Disables the option
      */
     disabled: boolean;
     /**
-     * Allows to show a tooltip to explain why the option is disabled
+     * Tooltip to display
      */
     tooltip: string;
   };
 }
 
+export interface SettingsMenuProps {
+  /**
+   * Custom props to populate the content of the settings meny
+   */
+  flyoutCustomProps?: FlyoutCustomProps;
+}
+
 /**
  * Renders a menu to allow the user to customize the flyout.
  * Current customization are:
  * - Flyout type: overlay or push
  */
 export const SettingsMenu: React.FC<SettingsMenuProps> = memo(
-  ({ flyoutTypeProps }: SettingsMenuProps) => {
+  ({ flyoutCustomProps }: SettingsMenuProps) => {
+    const dispatch = useDispatch();
+
+    // for flyout where the push vs overlay option is disable in the UI we fall back to overlay mode
+    const type = useSelector(selectPushVsOverlay);
+    const flyoutType = flyoutCustomProps?.pushVsOverlay?.disabled ? 'overlay' : type;
+
     const [isPopoverOpen, setPopover] = useState(false);
     const togglePopover = () => {
       setPopover(!isPopoverOpen);
     };
 
     const pushVsOverlayOnChange = useCallback(
       (id: string) => {
-        flyoutTypeProps.onChange(id as EuiFlyoutProps['type']);
+        dispatch(
+          changePushVsOverlayAction({
+            type: id as EuiFlyoutProps['type'] as 'overlay' | 'push',
+            savedToLocalStorage: !flyoutCustomProps?.pushVsOverlay?.disabled,
```

Comment:
is `savedToLocalStorage` and middleware still necessary with the new set up? i could see adding a `localStorage.setItem` here should suffice. wdyt?